### PR TITLE
Add Cabal Helper cradle and tests

### PR DESCRIPTION
**Pull request by fendor**
Vastly copied from https://github.com/haskell/haskell-ide-engine/

Copies tests and the cabal-helper cradle, does not modify any existing behaviour.

### CODE REVIEW

File: implicit-hie.cabal [hunk 1]
```diff
@@ -1,11 +1,4 @@
-cabal-version: 2.0
-
--- This file has been generated from package.yaml by hpack version 0.33.0.
---
--- see: https://github.com/sol/hpack
---
--- hash: 18f92037a7863d121ac45e847f7dc6177adf0ebc7951dfa1588340f86e64456b
-
+cabal-version: 2.4
```
Comment (fendor) [inline implicit-hie.cabal:1]: useful so that the extra-source-files dont need to be as verbose

**Comment by Avi-D-coder** [comment]
Wow, that was fast!
I'll take a look later today.

File: implicit-hie.cabal [hunk 2]
```diff
@@ -16,41 +9,68 @@ bug-reports:    https://github.com/Avi-D-coder/implicit-hie/issues
...
     , base >=4.7 && <5
+    , containers
     , directory >= 1.3
     , filepath >= 1.4
     , filepattern >= 0.1
+    , hie-bios >=0.5 && <0.6
+    , hslogger
     , text >= 1.2
     , transformers >= 0.5
+    , process
```
Comment (Avi-D-coder) [inline implicit-hie.cabal:68]: All these new deps should be behind the flag.

Comment (fendor) [reply implicit-hie.cabal]: Done.
I argue though that the hslogger dependency is fine, since implicit-hie needs a logger anyway.

Comment (Avi-D-coder) [reply implicit-hie.cabal]: Sounds good

**Comment by fendor** [comment]
> It should be possible to select cabal-helper programmatically in the lib and via a command line flag a la #3.

It is possible to use the cradle via the lib. I do not want to implement any visible behaviour with this PR, the follow up PR #11 should implement other changes

**Comment by fendor** [comment]
Closed as it seems like we want to go a different way

**Pull request closed without merging**